docs/libcurl/curl_easy_init.3

```diff
@@ -58,8 +58,16 @@ if(curl) {
 .SH AVAILABILITY
 Always
 .SH RETURN VALUE
-If this function returns NULL, something went wrong and you cannot use the
-other curl functions.
+Returns a valid non-null handle on success.
+
+On failure, returns NULL. If \fIcurl_global_init(3)\fP was called prior to
+\fIcurl_easy_init(3)\fP, then it means the process has ran out of memory. If
+\fIcurl_global_init(3)\fP was not called beforehand, it could also be due to a
+global initialization failure.
```

Comment:
I don't think it should be this specific. We don't want to exclude the possibility that this function can fail differently in a future.

Reply:
How is the user supposed to know when that was written and if their current version is such a future version? They probably are already using a future version when they read that and then having the "future version" text is superfluous and this docs is mostly phrased in a peculiar way.

Reply:
By the way, throughout this discussion there's an underlying assumption that out-of-memory errors are rare. Although that has been true in recent history with memory getting bigger and cheaper, that's only true when the software has full reigns of the machine.

In a containerized world, it's increasingly common to enforce soft and hard memory limits, specially on 3rd-party services and agents. I've seen first hand cases where debugging time lowers dramatically by having an out-of-memory error code vs an uninformed crash. That being said, there are other ways to detect out-of-memory, yet detecting by proxy remains less powerful than detecting _in loco_.

That said, it seems that the docs are already in the intended state and there's no interested in the proposed change, so I'll just drop this PR.

Thank you all for your time reviewing and sharing your experience as maintainers.

Reply:
curl is used in a *wide* variety of machines, environments and circumstances. Some of them with tiny system resources. I don't think out-of-memory is rare in all those instances. I don't see how this is very relevant to this PR though.